snuba/snuba_migrations/generic_metrics/0029_add_use_case_id_index.py

```diff
@@ -0,0 +1,76 @@
+from typing import Sequence
+
+from snuba.clusters.storage_sets import StorageSetKey
+from snuba.migrations import migration, operations
+
+
+class Migration(migration.ClickhouseNodeMigration):
+    blocking = False
+
+    def forwards_ops(self) -> Sequence[operations.SqlOperation]:
+        return [
+            operations.AddIndex(
+                storage_set=StorageSetKey.GENERIC_METRICS_COUNTERS,
+                table_name="generic_metric_counters_aggregated_local",
+                index_name="set_idx_use_case_id",
+                index_expression="use_case_id",
+                index_type="set(0)",
+                granularity=100,
```

Comment:
The granularity on the indexes is a multiple on the original granularity of the table. In this case, the table has a granularity of 2048. If you set this granularity to 100, this index's granularity would become (2048 * 100 = 204800). I think what you would want here is granularity of 1. So that the set only contains unique use cases for 2048 rows. Higher the number of rows for the index, the more useless it would become.